app/views/exercises/show.html.erb

```diff
@@ -65,6 +65,6 @@
     <%= render partial: 'layouts/modals/level_up' %>
   <% end %>
 
-  <%= render partial: 'layouts/modals/guide_corollary', locals: {guide: @guide} %>
+  <%= render partial: 'layouts/modals/guide_corollary', locals: {guide: @guide} if @stats.almost_done? %>
```

Comment:
This is a very simple optimization that avoid generating this partial in most scenarios. The `ExercisesSolutionsController` will benefit from this in future PRs too.